torchtitan/experiments/kernels/moe/indices.py

```diff
@@ -170,19 +186,27 @@ def generate_permute_indices(
         To:   | E0 | E1 | E2 | E3 | E0 | E1 | E2 | E3 |
               |  4 |  2 |  1 |  3 |  1 |  2 |  3 |  4 |
     """
+
     # prefix sum to get start index of each expert (parallel scan kernel in future?)
     start_index_values = (
         torch.cumsum(tokens_per_expert_group, 0) - tokens_per_expert_group
     )
 
-    # chunk sizes for each expert
-    chunk_size_per_expert = tokens_per_expert_group.view(num_ranks, -1).sum(0)
```

Comment:
How about:
```
is_zero = chunk_size_per_expert == 0
chunk_size_per_expert += is_zero.int()
```
It seems that's all we need to fix the issue.

Reply:
thanks! definitely agree that we can get rid of the intermediate tensor aspect, but also think we can make this even cleaner with just using torch.where instead of the .int() conversion.
~~~
padded_total_tokens_per_expert = torch.where(
        total_tokens_per_expert == 0, min_slots_per_expert, total_tokens_per_expert
    )
   ~~~ 
   works well. 
  Keeping the padded count and original count also means the skip logic can be used in the kernel. 
   
I'm not clear if we want to expose an actual min_slots_per_expert or simply pad up to m_align_size and that's sufficient. (i.e. for fp8, we need to use 16 for this min_slots but not sure if that is also the same for min alignment for fp8? )

Reply:
actually imo it's cleaner just to pad up zero experts to alignment and that should suffice. 
Thus getting rid of min_slots_per_token and use alignment to control things directly re: padding.